luigi-tutorials/tutorial-container.md

```diff
@@ -0,0 +1,294 @@
+```javascript
+HTMLElement.prototype.attachInvalidate = function() {};
+
+sap.ui.define([
+    "sap/ui/core/webc/WebComponent",
+    "@luigi-project/container"
+], function(WebComponent) {
+    "use strict";
+
+    /**
+     * Constructor for a new <code>LuigiContainer</code>.
+     *
+     * @param {string} [sId] ID for the new control, generated automatically if no ID is given
+     * @param {object} [mSettings] Initial settings for the new control
+     *
+     * @extends sap.ui.core.webc.WebComponent
+     * @class
+     *
+     * The <code>luigi.ui5app.control.LuigiContainer</code> is a component to render a Luigi micro frontend (iframe or web component based) without the need of being a Luigi Core application.
+     *
+     * @author SAP SE
+     *
+     * @constructor
+     * @public
+     * @alias luigi.ui5app.control.LuigiContainer
+     */
+    var LuigiContainer = WebComponent.extend("luigi.ui5app.control.LuigiContainer", {
+        metadata: {
+            tag: "luigi-container",
+            properties: {
+                /**
+                 * Defines the view URL to load
+                 */
+                viewURL: {
+                    type: "string",
+                    defaultValue: "",
+                    mapping: {
+                        to: "viewURL"
+                    }
+                },
+
+                /**
+                 * Defines whether to load the Luigi component as web component
+                 */
+                webComponent: {
+                    type: "any", // boolean, but WebComponentRenderer renders out boolean values as empty attributes, which we have to fix in our attribute formatter
+                    defaultValue: false,
+                    mapping: {
+                        to: "webcomponent",
+                        formatter: "_mapIsWebComponent"
+                    }
+                },
+
+                /**
+                 * Defines the height of the control
+                 */
+                height: {
+                    type: "sap.ui.core.CSSSize",
+                    mapping: "style"
+                },
+
+                /**
+                 * Defines the width of the control
+                 */
+                width: {
+                    type: "sap.ui.core.CSSSize",
+                    mapping: "style"
+                },
+
+                /**
+                 * Defines label of the Luigi component
+                 */
+                label: {
+                    type: "string",
+                    defaultValue: ""
+                },
+
+                /**
+                 * Defines the context to use
+                 */
+                context: {
+                    type: "object",
+                    defaultValue: {}
+                },
+
+                theme: {
+                    type: "string",
+                    defaultValue: ""
+                }
+            },
+            methods: ["updateContext", "sendCustomMessage", "closeAlert"]
```

Comment:
I meant, you can comment the methods here - not adding this generic comment at the bottom. 😉

Reply:
Thank you! I'm not very familiar with JavaScript comment conventions, but I removed most of the extra comments now and the file is a lot shorter.